llvm/test/Transforms/InstCombine/icmp-of-xor-x.ll

```diff
@@ -5,6 +5,327 @@ declare void @llvm.assume(i1)
 declare void @barrier()
 declare void @use.i8(i8)
 
+; X s< ~X --> X s< 0
+define i1 @src_slt_xnx_to_slt_0_i1(i1 %x) {
```

Comment:
I think it is enough to keep `i8` tests and some `i128` tests. Other tests are redundant.

Reply:
consider it done!

Reply:
Please add some tests that commute operands (e.g., `icmp sgt ~X, X`).
And some negative tests (e.g., `icmp sgt X, ~Y`).

Reply:
@dtcxzyw did you mean you want to add more fail tests? or you think it can be implement with this PR together?